core/src/main/scala/pureconfig/Path.scala

```diff
@@ -0,0 +1,9 @@
+package pureconfig
+
+final case class Path(segments: List[Path.Segment]) extends AnyVal
```

Comment:
Is there any advantage of having this class instead of using always a `Seq[Path.Segment]`? You could have a single `atPath` method if you were to remove this.